app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_db = {}
+
+    def wrapper(*args) -> any:
+        if args in cache_db:
+            print("Getting from cache")
+            return cache_db[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_db[args] = result
+            return result
```

Comment:
- what if the function takes the same args  but different kwargs
- you must use args  with kwargs as the key of cache_db

Reply:
Do I need to convert a dictionary to a tuple?